### ISSUE TITLE
ld: warning: ignoring duplicate libraries: '-lobjc'

### ISSUE THREAD

**Issue by prologic** [fyne-io/fyne#4502]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Running through the latest Fyne "Hello World" example I noticed this warning during build:

```console
$ go build .
# hello-fyne
ld: warning: ignoring duplicate libraries: '-lobjc'
```

### How to reproduce

1. Follow the [Getting Started](https://developer.fyne.io/started/) guide
2. Run `go build .`
3. Observe the ld warning

### Screenshots

<img width="1242" alt="Screenshot 2024-01-02 at 11 40 04" src="https://github.com/fyne-io/fyne/assets/1290234/410cc05a-3204-4696-9cfd-9bbffb2511fd">

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello World")

	w.SetContent(widget.NewLabel("Hello World!"))
	w.ShowAndRun()
}
```

### Fyne version

2.4.3

### Go compiler version

1.21.4

### Operating system and version

macOS Ventura 13.6.1

**Comment by r00tc0d3** [comment]
Same here

**Comment by andydotxyz** [comment]
This seems to be related to Go 1.21 not Sonoma (as it does not happen with 1.20 or earlier)

**Comment by CryoByte33** [comment]
I am having the same issue on MacOS 14.3.1, Go version 1.22 and Fyne version 2.4.4. 

I found [this thread](https://stackoverflow.com/questions/77164140/ld-warning-ignoring-duplicate-libraries-lgcc-after-the-recent-update-of-xc) and [this post](https://indiestack.com/2023/10/xcode-15-duplicate-library-linker-warnings/) that said it's a MacOS Sonoma linker change, and [this one](https://github.com/orgs/Homebrew/discussions/4794#discussioncomment-7044468) that lists it as a possible homebrew issue, but neither the linker flags or reinstalling every brew package on my system was able to get my Fyne project to build.

**Comment by andydotxyz** [comment]
It is just a warning and will not have stopped your project from building (that's partly why it's not been a high priority).
If you had a build error it was something else in the output than this which was the problem.

**Comment by skplunkerin** [comment]
Same issue here on my Apple M2 Pro:

**MacOS Version:**
Sonoma 14.3 (23D56)

**Go Version:**
go1.21.4 darwin/arm64

**Fyne Version:**
v2.4.4

**Comment by andydotxyz** [comment]
Just confirming - for you too it is just a warning and does not break the compilation or any functionality?

**Comment by skplunkerin** [comment]
> Just confirming - for you too it is just a warning and does not break the compilation or any functionality?

Correct, just an unnerving warning that doesn't prevent me from using Fyne :)

**Comment by CryoByte33** [comment]
> It is just a warning and will not have stopped your project from building (that's partly why it's not been a high priority). If you had a build error it was something else in the output than this which was the problem.

I replicated and you were right, it looks unrelated to my previous issue which was caused by another package's update. My bad! The message here persists without the flags posted above, though.

**Comment by XinCaoZ** [comment]
Same issue here on my Apple M3 Pro:

MacOS Version:
Sonoma 14.4 (23E214)

Go Version:
go1.22.1 darwin/arm64

Fyne Version:
v2.4.4

**Comment by XinCaoZ** [comment]
> Just confirming - for you too it is just a warning and does not break the compilation or any functionality?

But I hate the error message when I compile the project.😵‍💫

**Comment by andydotxyz** [comment]
Agreed, but I think the issue is in the Go compiler, so not sure what we can do.
"-lobjc" is added by the Go compiler not the Fyne source code...

**Comment by BenitoEck** [comment]
Still and issue on `go version go1.24.3 darwin/arm64` but does not really cause trouble.

**Comment by dmitshur** [comment]
The upstream Go tracking issue for this is https://github.com/golang/go/issues/67799.

**Comment by andydotxyz** [comment]
I've pinged on that issue to see where it's at / if we can help
